### ISSUE TITLE
"index_add_(): self and source must have the same scalar type"  while training DQN rainbow

### ISSUE THREAD

**Issue by tarscase** [iffiX/machin#4]
When i try to train the dqn rainbow agent i'm getting "index_add_(): self and source must have the same scalar type" error when weights are updated.

I'm using a simple Model to understand

    class QNet(nn.Module):
        # this test setup lacks the noisy linear layer and dueling structure.
        def __init__(self, action_num, atom_num=10):
            super(QNet, self).__init__()

            self.hidden_in = nn.Conv2d(4, 64, kernel_size = 4, stride = 2)
            self.fc1 = nn.Linear(105280, 64)
            self.fc2 = nn.Linear(64, 16)
            self.fc3 = nn.Linear(16, action_num * atom_num)
            self.action_num = action_num
            self.atom_num = atom_num
        
            self.flat = nn.Flatten()

        def forward(self, state):
            a = t.relu(self.hidden_in(state))
            a = self.flat(a)
        
            a = t.relu(self.fc1(a))
            a = t.relu(self.fc2(a))
            return t.softmax(self.fc3(a)
                             .view(-1, self.action_num, self.atom_num),
                         dim=-1)

i'm not unable to understand where is the tensor changing its type.
input is in shape (4,50,50) dtype.float32.

**Comment by iffiX** [comment]
I roughly know where this problem might occur, could you please show your complete setup?

**Comment by iffiX** [comment]
Should be fixed with https://github.com/iffiX/machin/commit/63b837a481f1a36b94734b8b1ef0701ca9c8e37c
